code/game/atoms.dm

```diff
@@ -159,6 +159,8 @@
 	var/damage_deflection = 0
 
 	var/resistance_flags = NONE // INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | ON_FIRE | UNACIDABLE | ACID_PROOF
+	/// forensics datum, contains fingerprints, fibres, blood_dna and hiddenprints on this atom
+	var/datum/forensics/forensics
```

Comment:
you need to handle this reference in atom `Destroy()`

i don't know off the top of my head if it should be `qdel`'d or just `null`ed, but you should probably check what the current behavior is (i think forensics gets qdel'd when its parent does, as it's a component)